be/src/storage/lake/horizontal_compaction_task.cpp

```diff
@@ -49,7 +50,8 @@ Status HorizontalCompactionTask::execute(Progress* progress) {
     reader_params.use_page_cache = false;
     RETURN_IF_ERROR(reader.open(reader_params));
 
-    ASSIGN_OR_RETURN(auto writer, _tablet->new_writer(kHorizontal));
+    int64_t txn_id = next_id();
```

Comment:
better use the actual txn id from FE.

Reply:
Uh ... how did I write next_id here ... I might have made a mistake when fixing the unit test

Reply:
Fixed